### PR TITLE
diagnostic: check all tap branches, not just core

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

This is relevant now that we have lots of Homebrew/cask taps that also need to have their branch checked, not just Homebrew/core.

### CODE REVIEW

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -566,20 +566,30 @@ def check_casktap_git_origin
...
+          next if tap.path.git_origin.nil?
 
-        branch = coretap_path.git_branch
-        return if branch.blank? || branch.include?("master")
+          branch = tap.path.git_branch
+          next if branch.blank?
 
-        <<~EOS
-          #{CoreTap.instance.full_name} is not on the master branch.
+          origin_branch = Utils::Git.origin_branch(tap.path)&.split("/")&.last
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb:581]: This is a bit hard to follow and feels a bit brittle. Might be worth being a dedicated (tested) function on `Utils::Git` or `Tap`

File: Library/Homebrew/diagnostic.rb [hunk 2]
```diff
@@ -566,20 +566,30 @@ def check_casktap_git_origin
...
+          next if origin_branch == branch
+
+          "git -C $(brew --repo #{tap.name}) checkout #{origin_branch}"
+        end.compact
 
-          Check out the master branch by running:
-            git -C "$(brew --repo homebrew/core)" checkout master
+        return unless commands.presence
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb:587]:
```suggestion
        return if commands.blank?
```

File: Library/Homebrew/diagnostic.rb (hunk 1 above)
Comment (jonchang) [reply Library/Homebrew/diagnostic.rb:581]: I have a refactor for this written already but didn't open it yet because it would have caused trouble for this pull request.

Comment (MikeMcQuaid) [reply Library/Homebrew/diagnostic.rb:581]: Fine with me if there's a refactor incoming 👍🏻
